frontend/src/model/beta/settings/ModelAccess.tsx

```diff
@@ -0,0 +1,177 @@
+
+  useEffect(() => {
+    if (model) {
+      setAccessList(model.collaborators)
+    }
+  }, [model, setAccessList])
+
+  function onUserChange(_event: React.SyntheticEvent<Element, Event>, newValue: User | null) {
+    if (
+      newValue &&
+      accessList.find(({ entity }) => entity === `user:${newValue.id}`) === undefined &&
+      newValue &&
+      accessList.find(({ entity }) => entity === `group:${newValue.id}`) === undefined
+    ) {
+      const updatedAccessList = accessList
+      const newAccess = { entity: `user:${newValue.id}`, roles: ['consumer'] }
+      updatedAccessList.push(newAccess)
+      setAccessList(accessList)
+    }
+  }
+
+  function updateAccessList() {
+    console.log(accessList)
+  }
+
+  return (
+    <>
+      {isUsersLoading && <Loading />}
+      {users && (
+        <Box sx={{ width: '750px' }}>
+          <Stack spacing={2}>
+            <Typography variant='h6' component='h2'>
+              Manage model access
+            </Typography>
+            <Autocomplete
+              open={open}
+              onOpen={() => {
+                setOpen(true)
+              }}
+              onClose={() => {
+                setOpen(false)
+              }}
+              // we might get a string or an object back
+              isOptionEqualToValue={(option: User, value: User) => option.id === value.id}
+              onChange={onUserChange}
+              getOptionLabel={(option) => option.id}
+              options={users || []}
+              renderInput={(params) => (
+                <TextField
+                  {...params}
+                  label='Add a user or group to the model access list'
+                  InputProps={{
+                    ...params.InputProps,
+                    endAdornment: (
+                      <>
+                        {isUsersLoading ? <Loading /> : null}
+                        {params.InputProps.endAdornment}
+                      </>
+                    ),
+                  }}
+                />
+              )}
+            />
+            <Box
+              sx={{
+                overflowY: 'auto',
+                maxHeight: '400px',
+                border: 'solid 1px',
+                padding: '20px',
+                borderColor: '#e0e0e0',
+                borderRadius: 1,
+              }}
+            >
+              <List disablePadding>
+                {accessList &&
+                  accessList.map((entity) => (
+                    <ListItem key={entity.entity}>
+                      <EntityItem entity={entity} accessList={accessList} setAccessList={setAccessList} />
+                    </ListItem>
+                  ))}
+              </List>
+            </Box>
+            <Button variant='outlined' aria-label='Save access list button' onClick={updateAccessList}>
+              Save
+            </Button>
+          </Stack>
+        </Box>
+      )}
+    </>
+  )
+}
+
+function EntityItem({
```

Comment:
The prevent files from becoming too large, consider pulling separate components like this into their own file. This also has to benefit of making unit testing individual components easier to organise as they can each have their own test file.